theme/models.py

```diff
@@ -293,6 +327,228 @@ def add_to_used_value(self, size):
+                                                  zone=self.zone,
+                                                  percent=rounded_percent)
+        else:
+            # return quota informational message
+            return_msg += qmsg.warning_content_prepend.format(allocated=self.allocated_value,
+                                                              unit=self.unit,
+                                                              used=rounded_used_val,
+                                                              zone=self.zone,
+                                                              percent=rounded_percent)
+        return return_msg
+
+
+class QuotaRequest(models.Model):
+    STATUS_CHOICES = (
+        ('pending', 'pending'),
+        ('denied', 'denied'),
+        ('approved', 'approved'),
+        ('revoked', 'revoked'),
+    )
+    status = models.CharField(max_length=20, choices=STATUS_CHOICES, default='pending')
+    request_from = models.ForeignKey(User, on_delete=models.CASCADE, related_name='ru2qrequest')
+    quota = models.ForeignKey(UserQuota, on_delete=models.CASCADE, related_name='g2qrequest')
+    date_requested = models.DateTimeField(editable=False, auto_now_add=True)
+    justification = models.TextField(validators=[MinLengthValidator(15, 'must contain at least 15 characters'),
+                                                 MaxLengthValidator(300, 'maximum 300 characters')], null=True)
+    storage = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(100)], default=1)
+
+    def notify_user_of_quota_action(self, send_on_deny=False):
+        """
+        Sends email notification to user on approval/denial of thie quota request
+
+        :param send_on_deny: whether emails should be sent on denial. default is to only send emails on quota approval
+        :return:
+        """
+
+        if self.status != 'approved' and not send_on_deny:
+            return
+
+        date = self.date_requested.strftime("%m/%d/%Y, %H:%M:%S")
+        email_msg = f'''Dear Hydroshare User,
+        <p>On { date }, you requested { self.storage } GB increase in quota.</p>
+        <p>Here is the justification you provided: <strong>'{ self.justification }'</strong></p>
+
+        <p>Your request for Quota increase has been reviewed and { self.status }.</p>
+
+        <p>Thank you,</p>
+        <p>The HydroShare Team</p>
+        '''
+        send_mail(subject=f"HydroShare request for Quota increase { self.status }",
+                  message=email_msg,
+                  html_message=email_msg,
+                  from_email=settings.DEFAULT_FROM_EMAIL,
+                  recipient_list=[self.request_from.email])
+
+
+class QuotaRequestForm(ModelForm):
+    class Meta:
+        model = QuotaRequest
+        fields = ["justification", "storage"]
+        labels = {
+            "justification": _("Enter justification for more quota"),
+            "storage": _("How much more storage to you require? (GB)"),
```

Comment:
Typo here. Change `to` to `do`.

Reply:
We are rendering this form using Django forms, which is not ideal, so I would suggest using plain HTML instead so that we can
1. Make the storage input smaller and align its text to the right (as is the standard for number inputs).
2. Display the unit next to the input by using [bootstrap ](https://getbootstrap.com/docs/3.3/components/#input-groups-basic)
![image](https://github.com/hydroshare/hydroshare/assets/2448568/deea4773-0c59-42f9-892a-0beddf447688)

Reply:
Great suggestions thank you. Added.